drop-storage/src/lib.rs

```diff
@@ -32,9 +32,50 @@ pub struct Storage {
 
 const MIGRATIONS_DIR: Dir = include_dir!("$CARGO_MANIFEST_DIR/migrations");
 
+#[cfg(unix)]
+fn prepare_sqlite_file(path: &str) -> io::Result<OpenFlags> {
+    use std::os::unix::prelude::{OpenOptionsExt, PermissionsExt};
+
+    const MODE: u32 = 0o600;
+    let mut flags = OpenFlags::default();
+
+    if path == ":memory:" {
+        return Ok(flags);
+    }
+    // disalbe file creation since we are going to do it ourselves
+    flags &= !OpenFlags::SQLITE_OPEN_CREATE;
+
+    match std::fs::metadata(path) {
+        Ok(meta) => {
+            let mut perm = meta.permissions();
+            if perm.mode() != MODE {
+                perm.set_mode(MODE);
+                std::fs::set_permissions(path, perm)?;
+            }
+        }
+        Err(err) if err.kind() == io::ErrorKind::NotFound => {
+            // create new one
+            std::fs::OpenOptions::new()
+                .write(true)
+                .create_new(true)
+                .mode(MODE)
+                .open(path)?;
+        }
+        Err(err) => return Err(err),
+    }
+
+    Ok(flags)
+}
```

Comment:
We should include a test case for permissions if we alter the existing one as well, just in case. Initially, I thought about just affecting newly created database permissions and leaving existing ones as it is as it's additional problem vector. What is your take on this?

Reply:
Yeah. I can add those test cases

Reply:
done